gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/EncounterMapper.java

```diff
@@ -0,0 +1,327 @@
+            .stream()
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(compoundStatement -> compoundStatement != null && compoundStatement.getClassCode().get(0).equals(TOPIC_CLASS_CODE))
+            .toList();
+    }
+
+    private Encounter mapToEncounter(RCMRMT030101UK04EhrComposition ehrComposition, Patient patient) {
+        var id = ehrComposition.getId().getRoot();
+
+        var encounter = new Encounter();
+        encounter
+            .setParticipant(getEncounterParticipant(ehrComposition.getAuthor(), ehrComposition.getParticipant2()))
+            .setStatus(EncounterStatus.FINISHED)
+            .setSubject(new Reference(patient))
+            .setType(getEncounterType(ehrComposition.getCode()))
+            .setPeriod(getPeriod(ehrComposition.getEffectiveTime(), ehrComposition.getAvailabilityTime()))
+            .setIdentifier(getEncounterIdentifier(id))
+            .setMeta(getEncounterMeta())
+            .setId(id);
+
+        setEncounterLocation(encounter, ehrComposition);
+
+        return encounter;
+    }
+
+    private List<CodeableConcept> getEncounterType(CD code) {
+        return List.of(codeableConceptMapper.mapToCodeableConcept(code));
+    }
+
+    private Meta getEncounterMeta() {
+        return new Meta().addProfile(ENCOUNTER_META_PROFILE);
+    }
+
+    private List<Identifier> getEncounterIdentifier(String id) {
+        Identifier identifier = new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+        return List.of(identifier);
+    }
+
+    private Period getPeriod(IVLTS effectiveTime, TS availabilityTime) {
+        Period period = new Period();
+        var center = getTSStringValue(effectiveTime.getCenter());
+        var low = getIVXBTSStringValue(effectiveTime.getLow());
+        var high = getIVXBTSStringValue(effectiveTime.getHigh());
+        var availabilityTimeValue = availabilityTime.getValue();
+
+        if (validValue(center)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(center));
+        } else if (validValue(low) && validValue(high)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(low))
+                .setEndElement(DateFormatUtil.parseToDateTimeType(high));
+        } else if (validValue(low) && !validValue(high)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(low));
+        } else if (!validValue(low) && validValue(high)) {
+            return period.setEndElement(DateFormatUtil.parseToDateTimeType(high));
+        } else if (CsNullFlavor.UNK.value().equals(center)) {
+            return null;
+        } else if (validValue(availabilityTimeValue)) {
+            return period.setStartElement(DateFormatUtil.parseToDateTimeType(availabilityTimeValue));
+        }
+
+        return null;
+    }
+
+    private String getTSStringValue(TS ts) {
```

Comment:
Might want to check with Alan but you're both doing the same logic here so perhaps one of you wants to make it reusable

Reply:
Alan's has added getters comparators to this TS class, so i have used the same logic, however our logic is slightly different so there is no needs for a shared Util for this.